spec/function.dd

```diff
@@ -1238,9 +1238,13 @@ int foo(in int x, out int y, ref int z, int q);
     for its type)
 
     $(TROW $(D ref),   parameter is passed by reference)
-    $(TROW $(D scope), references in the parameter
+    $(TROW $(D scope), References to the parameter
     cannot be escaped (e.g. assigned to a global variable).
-    Ignored for parameters with no references)
+    Enforced only for `@safe` functions.
+    Ignored for parameters with no references.
+    Note: The `scope` attribute is currently only enforced by `dmd`
+    when the `-dip1000` switch is passed.
```

Comment:
Doesn't `scope` on delegate has some affect even without `-dip1000`?

Reply:
Delegates are reference types though. But the elision of GC allocation of the closure should be mentioned.

Reply:
@jacob-carlborg I've now documented this, in a more general way. Walter has said that heap allocation for other reference types besides delegates can be elided too, e.g. for array concatenation:
https://forum.dlang.org/post/ofo9f5$2bt7$1@digitalmars.com